Add gradle configuration for static code analyzer

f2rf2r opened the pull request.

prfarlow1 commented:
A couple observations:
CompareObjectsWithEquals seems helpful but it does flag View comparison using ==
I'm not a fan of the UselessParentheses rule; I think sometimes they add clarity

prfarlow1 commented on app/static-analysis.gradle:
```diff
@@ -8,6 +8,8 @@ task pmd(type: Pmd) {
     exclude '**/gen/**'
 
+
+
```
extra whitespace

gsintrepid commented:
I agree with @prfarlow1, in Stetson there were 23 "UselessParentheses" warnings and many were useful for several/long comparisons.
The other warnings made sense!

deanshi commented:
For the PMD, is there a reason why we're not using the Empty ruleset?

f2rf2r commented:
@deanshi I just forgot/overlooked it. I have added that in the gradle plugin pr. I also changed this project to use the plugin now, so future comments regarding the rulesets should move to that repo. https://github.com/IntrepidPursuits/static-analysis-gradle-plugin/pull/1

deanshi commented on app/build.gradle:
```diff
@@ -1,3 +1,7 @@
+plugins {
+    id "io.intrepid.static-analysis" version "1.0-alpha6"
```
Will this be updated to 1.0 to match the README of the `static-analysis-gradle-plugin` once it's merged in?

f2rf2r replied:
yes
